connman/src/wispr.c

```diff
@@ -904,9 +936,35 @@ static bool wispr_portal_web_result(GWebResult *result, gpointer user_data)
 	return false;
 }
 
+static char *parse_proxy(const char *proxy)
+{
+	char *proxy_server;
+
+	if (!g_strcmp0(proxy, "DIRECT"))
+		return NULL;
+
+	if (!g_str_has_prefix(proxy, "PROXY "))
+		return NULL;
+
+	proxy_server = g_strdup(proxy + 6);
+
+	/* Use first proxy server */
+	for (char *c = proxy_server; *c != '\0'; ++c) {
```

Comment:
there are standard functions like strchr(), strcspn(), strpbrk(), etc that could be used for finding 1st instance of ';' instead of a custom loop...

Reply:
Didn't pay much attention to this upstream change. But yeah, that could be changed indeed.

Reply:
Like this 36e30b88e6f01782ae8a669fed4bb62212d8bafa